src/common/core/utils.py

```diff
@@ -84,6 +82,17 @@ def get_version_info() -> VersionInfo:
     return version_json
 
 
+@lru_cache()
+def get_versions_from_manifest() -> dict[str, str] | None:
+    """Reads the version info baked into the Docker container"""
+    raw_content = get_file_contents(VERSIONS_INFO_FILE_LOCATION)
+    if not raw_content:
+        return None
+
+    manifest: dict[str, str] = json.loads(raw_content)
+    return manifest
```

Comment:
If we return a default manifest, we'd lose a lot of boilerplate in the middleware:

```suggestion
    if not raw_content:
        return {".": "unknown"}
    manifest: dict[str, str] = json.loads(raw_content)
    return manifest
```

Then, in `middleware.py`:

```python
    def __call__(self, request: HttpRequest) -> HttpResponse:
        response = self.get_response(request)
        response.headers["X-Flagsmith-Version"] = get_versions_from_manifest()["."]        
        return response
```